toolchest_client/tools/api.py

```diff
@@ -168,6 +168,61 @@ def kraken2(output_path, inputs=[], database_name="standard", database_version="
     instance.run()
 
 
+def megahit(output_path, tool_args="", read_one=None, read_two=None, interleaved=None,
+            single_end=None):
+    """Runs Megahit via Toolchest.
+
+    :param output_path: Path (client-side) where the output will be downloaded.
+    :param tool_args: (optional) Additional arguments to be passed to Megahit.
+    :param read_one: (optional) `-1` inputs. Path or list of paths for read 1 of paired-read input files.
+    :param read_two: (optional) `-2` inputs. Path or list of paths for read 2 of paired-read input files.
+    :param interleaved: (optional) `--12` inputs. Path or list of paths for interleaved paired-end files.
+    :param single_end: (optional) `-r` inputs. Path or list of paths for single-end inputs.
+
+    .. note:: Each read in `read_one` should match with a read in `read_two`, and vice
+    versa. In other words, the nth read in `read_one` should be paired with the nth read
+    in `read_two`.
+
+    Usage::
+
+        >>> import toolchest_client as toolchest
+        >>> toolchest.megahit(
+        ...     tool_args="",
+        ...     read_one=["./pair_1/r1.fa", "./pair_2/r1.fa"],
+        ...     read_two=["./pair_1/r2.fa", "./pair_2/r2.fa"],
+        ...     output_path="./path/to/output.tar.gz",
+        ... )
+
+    """
+
+    # If input parameters are lists, parse these for input_prefix_mapping.
+    tag_to_param_map = {
+        "-1": read_one,
+        "-2": read_two,
+        "--12": interleaved,
+        "-r": single_end,
+    }
+    input_list = []  # list of all inputs
+    input_prefix_mapping = {}  # map of each input to its respective tag
+    for tag, param in tag_to_param_map.items():
+        if isinstance(param, list):
+            for input_file in param:
+                input_list.append(input_file)
+                input_prefix_mapping[input_file] = tag
+        elif isinstance(param, str):
+            input_list.append(param)
+            input_prefix_mapping[param] = tag
```

Comment:
This is more a question than a comment.

Our code in api.py file is getting more and more complex. I wonder if there's a better place for logic like this?